Src/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -1719,12 +1719,18 @@ public static class StringOps {
         private static DecoderFallback ReplacementFallback = new DecoderReplacementFallback("\ufffd");
 #endif
 
-        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e) => DoDecode(context, s, errors, encoding, e, true, out _);
+        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e)
+            => DoDecode(context, s, errors, encoding, e, true, true, out _);
 
-        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e, bool final, out int numBytes) {
+        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e, bool final, out int numBytes)
+            => DoDecode(context, s, errors, encoding, e, true, final, out numBytes);
+
+        // If the beginning of input s is the same as the preamble in Encoding e, those bytes are never decoded or passed to output.
+        // However, if discountPreamble is false, the number of those bytes is included in total numBytes reported.
+        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e, bool discountPreamble, bool final, out int numBytes) {
```

Comment:
Is there a use case for having `discountPreamble`? Just looking at the usage, when `discountPreamble == true` we always end up discarding `numBytes`. Unless you have specific plans for `discountPreamble` I would just remove it and use the `false` behavior all the time.

Reply:
I had no plans for further use of `discountPreamble`. Quite the opposite: I've made a task to to for myself to look into getting rid of it. I found it strange that while the functions I was working on require the preamble bytes to be counted, the default behaviour of `DoDecode` is to discount them. I thought it was either a bug or maybe those few cases where it is really needed can be handled locally. 

I did not look into the effect of always counting the preamble bytes, thinking this may be difficult to trace (and also because so many codecs tests are still disabled), but I am encouraged by your observation that by default we discard `numBytes` anyway. So I will do some checks on my side as well and remove `discountPreamble` in the next commit.